### PR TITLE
[MNT] Refactored `DescriptionEditor` component to auto-save

### LINKED ISSUE #200: Refactor `DescriptionEditor` component
At the moment the component has a button to enable user to edit the description and then save the description. We want to refactor so:
- it no longer requires the user to click the edit button and they just click the field and start typing
- it automatically saves the description as the user types in the text field maybe with an indication (like a text saying saved...).

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #200 

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [ ] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

## Summary by Sourcery

Refactor DescriptionEditor to auto-save edits with a 500ms debounce and status indicator, remove explicit edit/save buttons, and update tests to validate the new behavior

New Features:
- Automatically save description edits with debounce and display saving/saved status
- Show placeholder text when no description is provided

Enhancements:
- Consolidate editing UI into a single TextField and sync local state on prop changes
- Convert empty input to null before saving

Tests:
- Revise e2e and component tests to interact via typing and assert auto-save instead of clicking save button
- Add tests for placeholder display, auto-save callback invocation, and empty-input handling

### CODE REVIEW

File: src/components/DescriptionEditor.tsx [hunk 1]
```diff
@@ -23,66 +21,69 @@ function DescriptionEditor({
...
 
-  const handleEditDescription = () => {
-    setIsEditingDescription(true);
-  };
+  // Update local state when description prop changes
+  useEffect(() => {
+    setEditedDescription(description);
+  }, [description]);
+
+  // Debounced save function
```
Comment (surchs) [inline src/components/DescriptionEditor.tsx:34]:
```suggestion
  // This function ensures that we only update the description after the user stops typing
  //  When a user types, start a timer before storing the current value. If they type again before the timer expires, create a new timer and begin waiting again
```

File: src/components/DescriptionEditor.tsx [hunk 2]
```diff
@@ -23,66 +21,69 @@ function DescriptionEditor({
...
-  const [isEditingDescription, setIsEditingDescription] = useState(false);
   const [editedDescription, setEditedDescription] = useState<string | null>(description);
+  const [saveStatus, setSaveStatus] = useState<'idle' | 'saving' | 'saved'>('idle');
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
   const dataCy = levelValue ? `${columnID}-${levelValue}` : columnID;
 
-  const handleEditDescription = () => {
-    setIsEditingDescription(true);
-  };
+  // Update local state when description prop changes
```
Comment (surchs) [inline src/components/DescriptionEditor.tsx:29]:
```suggestion
```

**Review by surchs — APPROVED** [review]
Thanks @rmanaem, looks good and works well.

Left a suggestion.

🧑‍🍳
